### ISSUE TITLE
PR these packages upstream?

### ISSUE THREAD

**Issue by WinterMute** [proconsule/nxmp-portlibs#8]
It would be great if you could PR the packages you need upstream to https://github.com/devkitPro/pacman-packages & then everyone gets the benefit. It's not ideal to have packages with different dependencies & compile options in different places.

**Comment by proconsule** [comment]
Yes,i will do it, but some packages like ffmpeg is a bit different from the stock (libsmb2 support,libns support) and need to be well tested.
Also some libs cannot be build on windows toolchain and i am preparing a bunch of issues with current windows toolchain to describe you the problems (meson build is the worst)

For sure i will pr the PKGBUILD and patch for building all the libs needed by nxmp to be built, i  also like to have a unified repo, but since the video scene is movong fast this months, thanks to averne tx1 decoder, not all parts are ready to mass distribution.

As side note. I want to thank you and all the devkitpro dev for the effort to bring the best toolchain possible for hombrew devs and i am very happy to contribute to it!

**Comment by WinterMute** [comment]
Building things on windows can be a bit of a pain. We package everything from linux mostly for that reason as well as the issues with permissions on packages. It can be worth having a look at the packages done by Msys2 for pointers although they're also patching libraries for use on windows. https://github.com/msys2/MINGW-packages

**Comment by proconsule** [comment]
Yes i know that windows may lead to problems and need workaround to get it working, but at least meson build have to be implemented. i will look at MINGW packages for what is mandatory for meson cross compile on windows and let you know.

**Comment by WinterMute** [comment]
fwiw we have a couple of packages that use meson & we have a helper script to construct a crossfile.txt for building. We're currently only using it for switch but it will support all the consoles we target.

See https://github.com/devkitPro/pacman-packages/blob/master/switch/dav1d/PKGBUILD#L29 & https://github.com/devkitPro/pacman-packages/blob/master/switch/mesa/PKGBUILD#L32

Does that help any?

**Comment by proconsule** [comment]
this is the main problem of meson on windows 

`ERROR: This python3 seems to be msys/python on MSYS2 Windows, but you are in a MinGW environment

ERROR: Please install and use mingw-w64-x86_64-python3 and/or mingw-w64-x86_64-meson with Pacman
`

A missing packages in pacman for mingw prevents the start of crosscompile

**Comment by WinterMute** [comment]
> this is the main problem of meson on windows
> 
> `ERROR: This python3 seems to be msys/python on MSYS2 Windows, but you are in a MinGW environment
> 
> ERROR: Please install and use mingw-w64-x86_64-python3 and/or mingw-w64-x86_64-meson with Pacman `
> 
> A missing packages in pacman for mingw prevents the start of crosscompile

The problem is that you're in a MinGW environment as the initial message says. Install python & meson from the msys2 repo - i.e. `pacman -S meson python`. Run makepkg from an msys2 shell, i.e. start->devkitPro->MSys2.

**Comment by proconsule** [comment]
I started from msys and use makepkg and that was the error i got (tried libdav1).
I am in the process of migrating some stuff to different io system (devoptabs) so i nxmp can use a unmodified ffmpeg from averne repo.
For the meson build i will go deep to better reproduce my issues and report you what i found out.
Having a good windows buildchain is important for new dev to make samples and learn how it works.
For sure to go serious the linux  approach is the best.

**Comment by WinterMute** [comment]
> I started from msys and use makepkg and that was the error i got (tried libdav1).

I built libdav1d on msys2 just before I posted that reply to check & it worked fine. See  attached log.

If however I build it from a mingw64 shell then I get this
```
davem@ashpool MINGW64 ~/projects/devkitpro/pacman-packages/switch/dav1d
$ makepkg
==> Making package: switch-dav1d 1.2.1-1 (Mon Dec  4 22:38:36 2023)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading dav1d-1.2.1.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1442k    0 1442k    0     0  2220k      0 --:--:-- --:--:-- --:--:-- 2222k
  -> Found dav1d.patch
==> Validating source files with sha256sums...
    dav1d-1.2.1.tar.gz ... Passed
    dav1d.patch ... Passed
==> Extracting sources...
  -> Extracting dav1d-1.2.1.tar.gz with bsdtar
==> Starting prepare()...
patching file meson.build
Hunk #1 succeeded at 174 (offset -4 lines).
patching file src/cpu.c
patching file src/meson.build
Hunk #1 succeeded at 325 (offset -2 lines).
==> Starting build()...
ERROR: This python3 seems to be msys/python on MSYS2 Windows, but you are in a MinGW environment
ERROR: Please install and use mingw-w64-x86_64-python3 and/or mingw-w64-x86_64-meson with Pacman
==> ERROR: A failure occurred in build().
    Aborting...

davem@ashpool MINGW64 ~/projects/devkitpro/pacman-packages/switch/dav1d
$ uname -a
MINGW64_NT-10.0-19045 ashpool 3.4.9.x86_64 2023-09-15 12:15 UTC x86_64 Msys

```

Which is the error you're referring to. Run makepkg from an **Msys2** shell where uname says this

```
$ uname -a
MSYS_NT-10.0-19045 ashpool 3.4.9.x86_64 2023-09-15 12:15 UTC x86_64 Msys
```

You can also get there by running msys2.exe found in C:\devkitPro\msys2


[msys2-build.log](https://github.com/proconsule/nxmp-portlibs/files/13552145/msys2-build.log)

**Comment by proconsule** [comment]
I started msys as usual from start menu, but honestly i didnt veryfied with uname if that was a msys shell, probably not since my results.
I will make a test from a msys shell and let you know, but for sure will lead to your succesful result.

I see that new deko3d binary (i used it and works flawless) i have to check the ffmpeg build flags of current binary that may lack some features for hls streaming. 

2 libray will be very very important for me and also for the community.
libsmb2 and libnfs (the pkgbuild of this two can be found on averne repo) 

Thanks and sorry for the meson issue, i have to find out why my start link open a mingw64 shell instead msys

**Comment by WinterMute** [comment]
> I started msys as usual from start menu, but honestly i didnt veryfied with uname if that was a msys shell, probably not since my results. I will make a test from a msys shell and let you know, but for sure will lead to your succesful result.
> Thanks and sorry for the meson issue, i have to find out why my start link open a mingw64 shell instead msys

I think you probably followed the windows build instructions provided. I've made a PR to correct that for the time being. https://github.com/proconsule/nxmp-portlibs/pull/9
